### PR TITLE
[WIP] Renames Window to NvimWindow

### PR DESCRIPTION
**Pull request by joalon**
## Background
I tried to compile a project with libnvim and wayland-client libraries and hit upon a roadblock since both projects implement a Window datastructure. To get around this I renamed Window to NvimWindow. I posted this to see if anyone else has run into this and see what the maintainers think. I still think just naming it Window is cleaner but wanted to get some second opinions.

## Work done
I've renamed all instances I've found to NvimWindow, however I haven't touched any documentation or tests yet. Wanted to get this out there to gauge the interest.

### CODE REVIEW

File: src/nvim/api/private/defs.h [hunk 1]
```diff
@@ -107,7 +107,7 @@ typedef enum {
   kObjectTypeLuaRef,
   // EXT types, cannot be split or reordered, see #EXT_OBJECT_TYPE_SHIFT
   kObjectTypeBuffer,
-  kObjectTypeWindow,
+  kObjectTypeNvimWindow,
```
Comment (justinmk) [inline src/nvim/api/private/defs.h]: could be just `kNvimWindow`, I don't think there's any ambiguity there.

**Comment by justinmk** [comment]
I'm in favor of this. What do others think? @neovim/core

**Comment by norcalli** [comment]
Does that mean renaming every type like KeyValuePair, Integer, Dictionary, etc... to `Nvim$1`? That sounds fine for anything that's exported as a public type, but oof that's going to be a massive refactor. Also will that break existing things and how much do we care about that?

**Comment by bfredl** [comment]
An option would be to have `NvimWindow` etc in public libnvim headers (which does not yet exist, this would be done as part of _defining_ that interface), but typedef them to the existing names in internal headers and code, as to not cause massive churn.

File: src/nvim/api/private/defs.h (hunk 1 above)
Comment (justinmk) [reply src/nvim/api/private/defs.h]: ^ suggestion still stands. Full list of suggested names:

```
kNvimWin
kNvimTab
kNvimBuf
```

and for the types:

```
NvimWin
NvimTab
NvimBuf
```

File: src/nvim/api/private/helpers.h [hunk 2]
```diff
@@ -81,7 +81,7 @@
 #define api_init_float
 #define api_init_string = STRING_INIT
 #define api_init_buffer
-#define api_init_window
+#define api_init_nvimwindow
```
Comment (justinmk) [inline src/nvim/api/private/helpers.h]: I would not bother changing these helper macros unless we plan to add "nvim" prefix to all of them (`api_init_string` => `api_init_nvimstring`, ...). But that's probably unnecessary. At least wait until we have a reason to do it (e.g. build conflicts).

**Pull request closed without merging**